api/common/network.go

```diff
@@ -158,41 +158,33 @@ func updateParentForBridgePorts(bridgeName, sysClassNetPath string, nameToConfig
 	}
 }
 
-func interfaceAddressToNetworkConfig(interfaceName, configType string, address net.Addr) (params.NetworkConfig, error) {
+func interfaceAddressToNetworkConfig(
+	interfaceName, configType string, addr corenetwork.ConfigSourceAddr,
+) (params.NetworkConfig, error) {
 	config := params.NetworkConfig{
 		ConfigType: configType,
 	}
 
-	cidrAddress := address.String()
-	if cidrAddress == "" {
-		return config, nil
+	ip := addr.IP()
```

Comment:
Is there a guarantee that this function will never be called with a nil `addr` arg?

Reply:
Based on the current implementation, yes; but I will add a guard in any case.